packages/cache-inmemory/src/services/inmemory-cache.ts

```diff
@@ -0,0 +1,98 @@
+import {
+  ConfigurableModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
+  ICacheService,
+} from "@medusajs/medusa"
+import { MedusaError } from "medusa-core-utils"
+
+import { CacheRecord, InMemoryCacheModuleOptions } from "../types"
+
+const DEFAULT_TTL = 30 // seconds
+
+type InjectedDependencies = {}
+
+/**
+ * Class represents basic, in-memory, cache store.
+ */
+class InMemoryCacheService implements ICacheService {
+  protected TTL: number
+
+  protected readonly timoutRefs = new Map<string, number>()
+  protected readonly store = new Map<string, CacheRecord<any>>()
+
+  constructor(
+    deps: InjectedDependencies,
+    options: InMemoryCacheModuleOptions = {},
+    moduleDeclaration?: ConfigurableModuleDeclaration
+  ) {
+    this.TTL = options.defaultTTL || DEFAULT_TTL
+
+    if (moduleDeclaration?.resources !== MODULE_RESOURCE_TYPE.SHARED) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_ARGUMENT,
+        "At the moment this module can only be used with shared resources"
+      )
+    }
+  }
+
+  /**
+   * Retrieve data from the cache.
+   * @param key - cache key
+   */
+  async get<T>(key: string): Promise<T | null> {
+    const now = Date.now()
+    const record: CacheRecord<T> | undefined = this.store.get(key)
+
+    if (!record || (record.expire && record.expire < now)) {
+      return null
+    }
+
+    return record.data
+  }
+
+  /**
+   * Set data to the cache.
+   * @param key - cache key under which the data is stored
+   * @param data - data to be stored in the cache
+   * @param ttl - expiration time in seconds
+   */
+  async set<T>(key: string, data: T, ttl: number = this.TTL): Promise<void> {
+    const record: CacheRecord<T> = { data, expire: ttl * 1000 + Date.now() }
+
+    const oldRecord = this.store.get(key)
+
+    if (oldRecord) {
+      clearTimeout(this.timoutRefs.get(key))
+      this.timoutRefs.delete(key)
+    }
+
+    setTimeout(() => {
+      this.invalidate(key)
+    }, ttl * 1000)
+
```

Comment:
set it to `this.timoutRefs` is missing?

Reply:
Good catch, fixed in https://github.com/medusajs/medusa/pull/3187/commits/2284503bc5dc175f2aea36ce23362b41e4276d10